packages/transaction-controller/src/TransactionController.ts

```diff
@@ -672,11 +702,15 @@ export class TransactionController extends BaseControllerV1<
       origin,
     );
 
+    const ethQuery = this.getEthQuery(networkClientId);
+
     const transactionType =
-      type ?? (await determineTransactionType(txParams, this.ethQuery)).type;
+      type ?? (await determineTransactionType(txParams, ethQuery)).type;
 
     const existingTransactionMeta = this.getTransactionWithActionId(actionId);
-    const chainId = this.getChainId();
+    const chainId =
+      this.getNetworkClientById(networkClientId).configuration.chainId ??
+      this.getChainId();
```

Comment:
this should take networkClientId as an optional param instead of calling getNetworkClientById directly